.github/workflows/check_code.yml

```
@@ -0,0 +1,48 @@
+      - '**'
+    branches-ignore:
+      - main
+
+jobs:
+  lint-and-check:
+    name: Lint and Analyze Code
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
```

actionlint (1.7.4)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)